File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include<iostream>
using namespace std;

int main(){
    ios::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
    string s,bomb;
    string st;//스택처럼 쓸 문자열(push하듯이 글자 하나씩 붙여넣기 할거임.)
    cin>>s>>bomb;
    for(auto c : s){
        st+=c;
        if(st.size() >= bomb.size() && c==bomb[bomb.size()-1]){
            bool isBomb = true;
            for(int i = 1; i<=bomb.size(); i++){
                if(st[st.size()-i] != bomb[bomb.size()-i]) isBomb = false;
            }
            if(isBomb) st.erase(st.begin()+st.size()-bomb.size(), st.end());
        }
    }
    if(st.size())cout<<st;
    else cout << "FRULA";
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
s=input()
st =[]
b=input()
def isBomb():
    for i in range(1,len(b)+1):
        if st[-i] != b[-i]:
            return False 
    return True 


for i in s:
    st.append(i)
    if len(st) >= len(b) and i == b[-1] and isBomb():
        del st[-len(b):len(st)]

if len(st)==0:
    print("FRULA")
else:
    for i in st:
        print(i,end='')
